### ISSUE TITLE
Pact fails to rerun a request with multipart form data 

### ISSUE THREAD

**Issue by aniket-21** [DiUS/pact-jvm#376]
My tests generate a PACT file with multipart-form data. When I run the Pact file at provider end it fails saying "Stream ended unexpectedly".

If I send exactly same request as that of a PACT file by creating my own HTTP client, it works great. It seems something is messing up the request. Mostly I feel its the [this code](https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-provider/src/main/groovy/au/com/dius/pact/provider/ProviderClient.groovy#L110)

Here is my request from recorded interaction:  

"request": {
                "method": "POST",
                "path": "/some/path",
                "headers": {
                    "Content-Type": "multipart/form-data; boundary=----WebKitFormBoundary"
                },
                "body": "------WebKitFormBoundary\nContent-Disposition: form-data; name=\"file\"; filename=\"somemodel.txt\"\nContent-Type: application/octet-stream\nContent-Transfer-Encoding: binary\n\nsome model\n------WebKitFormBoundary--"
            }

Http Client request generated with exactly same body , works fine

**Comment by aniket-21** [comment]
There is kind of a deadlock. In above example if I manually edit request body and replace all "\n" with "\r\n" then provider test works.

To have "\r\n" in pact file, I need to provide it in PactDslJsonBody. If i provide body with "\r\n" it fails during consumer pact verification, reason being, by the time request hits mock server, all the "\r" are stripped off from request body and just "\n" exist (I am not sure how that happens)

Bottom line is provider test doesn't work **without "\r"** where ideally it should , and consumer pact verification doesn't work **with "\r\n"**

**Comment by aniket-21** [comment]
Finally found out the issue after 3 days of intensive debugging. This line of code was the culprit which was replacing "\r\n" with just "\n"
https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-consumer/src/main/scala/au/com/dius/pact/model/unfiltered/Conversions.scala#L71
